config.go

```diff
@@ -149,7 +150,7 @@ func (c *wafConfig) WithDebugLogger(logger debuglog.Logger) WAFConfig {
 	return ret
 }
 
-func (c *wafConfig) WithErrorCallback(logger func(rule types.MatchedRule)) WAFConfig {
+func (c *wafConfig) WithErrorCallback(logger func(context.Context, types.MatchedRule)) WAFConfig {
```

Comment:
This is one option, another one is to add a method for retrieving the context in MatchedRule cc @anuraaga

Reply:
I guess it could always be a different additional function, something like `WithContextErrorCallback(logger func(context.Context, types.MatchedRule)` if you didn't want to it to be a breaking change.

Reply:
It has to be a breaking change anyways because the WAF is an interface and we decided not too long ago that changing that will require breaking changes.

Reply:
ahh yes - I missed that, thanks.